src/widgets/unfold_widgets/preprocessing_widget.py

```diff
@@ -200,3 +207,35 @@ def remove_nulls_warning(self):
 
     def handle_warning_click(self, button):
         self.data_submitted = "OK" in button.text()
+
+    def reduce_dimensions(self, dim_number=None):
+        self.manual_reduction.clicked.disconnect()
+        self.auto_reduction.clicked.disconnect()
+        self.engine.reduce_dimensions(dim_number)
+        self.get_data()
+        self.show_reduction_results()
+
+    def render_reduction_group(self, max_dimensions):
+        for i in reversed(range(self.auto_reduction_group_layout.count())):
+            self.auto_reduction_group_layout.itemAt(i).widget().setParent(None)
+
+        self.num_dimensions_spinbox.setMinimum(2)
+        self.num_dimensions_spinbox.setMaximum(max_dimensions - 1)
+        self.num_dimensions_spinbox.setValue(2)
+        self.auto_reduction_group_layout.addRow(QLabel("Number of dimensions"), self.num_dimensions_spinbox)
+
+        self.manual_reduction.setText("Reduce with fixed number")
+        self.manual_reduction.setMinimumHeight(23)
+        self.auto_reduction_group_layout.addRow(self.manual_reduction)
+        self.manual_reduction.clicked.connect(lambda: self.reduce_dimensions(self.num_dimensions_spinbox.value()))
+        self.manual_reduction.setDisabled(max_dimensions < 3)
+
+        self.auto_reduction.setText("Reduce dynamically")
+        self.auto_reduction.setMinimumHeight(23)
+        self.auto_reduction_group_layout.addRow(self.auto_reduction)
+        self.auto_reduction.clicked.connect(lambda: self.reduce_dimensions())
+        self.auto_reduction.setDisabled(max_dimensions < 3)
```

Comment:
Maybe only change the range and value of spinbox instead of create new elements each time we click to preprocessing section.